src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_ISwitchStatement.cs

```diff
@@ -0,0 +1,1213 @@
+        }
+
+        [Fact, WorkItem(17600, "https://github.com/dotnet/roslyn/issues/17600")]
+        public void ISwitchStatement_GoverningExpressionWithLocalDeclarataion()
+        {
+            string source = @"
+using System;
+
+class Test
+{
+    void M()
+    {
+        /*<bind>*/switch (M2(out var number))
+        {
+            case 0:
+                break;
+        }/*</bind>*/
+    }
+
+    int M2(out int number)
+    {
+        number = 0;
+        return 1;
+    }
+}
+";
+            string expectedOperationTree = @"
+ISwitchStatement (1 cases) (OperationKind.SwitchStatement) (Syntax: 'switch (M2( ... }')
+  Switch expression: 
+    IInvocationExpression ( System.Int32 Test.M2(out System.Int32 number)) (OperationKind.InvocationExpression, Type: System.Int32) (Syntax: 'M2(out var number)')
+      Instance Receiver: 
+        IInstanceReferenceExpression (OperationKind.InstanceReferenceExpression, Type: Test, IsImplicit) (Syntax: 'M2')
+      Arguments(1):
+          IArgument (ArgumentKind.Explicit, Matching Parameter: number) (OperationKind.Argument) (Syntax: 'out var number')
+            ILocalReferenceExpression: number (IsDeclaration: True) (OperationKind.LocalReferenceExpression, Type: System.Int32) (Syntax: 'var number')
+            InConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+            OutConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+  Sections:
+      ISwitchCase (1 case clauses, 1 statements) (OperationKind.SwitchCase) (Syntax: 'case 0: ... break;')
+          Clauses:
+              ISingleValueCaseClause (CaseKind.SingleValue) (OperationKind.CaseClause) (Syntax: 'case 0:')
+                Value: 
+                  ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 0) (Syntax: '0')
+          Body:
+              IBranchStatement (BranchKind.Break) (OperationKind.BranchStatement) (Syntax: 'break;')
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyOperationTreeAndDiagnosticsForTest<SwitchStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact, WorkItem(17600, "https://github.com/dotnet/roslyn/issues/17600")]
+        public void ISwitchStatement_SwitchSectionWithLocalDeclarataion()
+        {
+            string source = @"
+using System;
+
+class Test
+{
+    int M(int number)
+    {
+        /*<bind>*/switch (number)
+        {
+            case 0:
+            case 1:
+                int number2 = 2;
+                return number2;
+        }/*</bind>*/
+
+        return 0;
+    }
+}
+";
+            string expectedOperationTree = @"
+ISwitchStatement (1 cases) (OperationKind.SwitchStatement) (Syntax: 'switch (num ... }')
```

Comment:
>ISwitchStatement [](start = 0, length = 16)

It looks like BoundSwitchStatement.InnerLocals and BoundSwitchStatement.InnerLocalFunctions are not exposed.